HDF5 reader and generation for gaussian basis wavefunctions

anbenali opened the pull request:
- HDF5 Converter hiding basis set and orbitals inside HDF5 file. 
- Reformating WFS file to look exactly like the Spline WFS.
- Allowing QMCPACK to read LCAO HDF5  
- Adding tests to ctests

prckent commented:
OK to test

prckent commented:
Some explanations are needed:

- Is this backwards compatible? e.g. Do all the tutorial/lab files work?
- At what point is using HDF5 worth it?
- Why was is necessary to update the files in tests/molecules?

anbenali commented:
- Yes it is backward compatible. Only targeted when using
Convert4qmc (...) -hdf5
It changes nothing but adds an option of having all the basis set and the
orbital stored in an hdf5 file instead of the very long xml file.

- I would say it is always useful for clarity and portability. But it is
definitely necessary if you use large molecules as the xml file becomes too
big.. 2M lines..

-  I wanted the WFS file for molecules to be identical to the one for
solids. In order to do so, I needed to switch "type" and "name" in the
definition of the the determinant set. Since the test are not based in xml
but simple diff in text file, swapping the tags led to a test fail when
technically the code does not care... so I changed the tests but in xml
language they are identical.




On Oct 10, 2017 20:28, "Paul R. C. Kent" <notifications@github.com> wrote:

Some explanations are needed:

   - Is this backwards compatible? e.g. Do all the tutorial/lab files work?
   - At what point is using HDF5 worth it?
   - Why was is necessary to update the files in tests/molecules?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/QMCPACK/qmcpack/pull/411#issuecomment-335654617>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AXl6w8b8UhBSpkF3lFQxD0b18Xrf1DVrks5srBm0gaJpZM4P0wYZ>
.

anbenali commented:
Do you want the converter to:
1- Do the conversion and print the recommendation message at the end of the
conversion (easiest way but with the risk of having people not reading)
2- Stop and ask " Do you prefer to use HDF5" ( I do not like this as it may
break scripts)
3- Generate both types of inputs? (possible but a bit more tedious)

On Wed, Oct 11, 2017 at 9:53 AM, Paul R. C. Kent <notifications@github.com>
wrote:

> *@prckent* requested changes on this pull request.
>
> Modify convert4qmc to print a suggestion to use the -hdf5 option if the
> molecule is "large" and the HDF5 route is not being taken. I suggest a
> heuristic based on the total number of MO coefficients in the file.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/QMCPACK/qmcpack/pull/411#pullrequestreview-68636796>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AXl6w3jXxi7a88F3xI8yMyeX9gvA_-Jcks5srNaFgaJpZM4P0wYZ>
> .
>



-- 
-----------------------
Anouar Benali, PhD
Leadership Computing Facility
Argonne National Laboratory
Building 240 Office - 2127
9700 S Cass Av., Argonne Il, 60439
(630) 252-0058

prckent commented:
Advice at the start

markdewing commented on src/QMCWaveFunctions/MolecularOrbitals/MolecularBasisBuilder.h:
```diff
@@ -62,62 +65,153 @@ class MolecularBasisBuilder: public BasisSetBuilder
+
+          EleTycBasisSet= H5Gopen(atomicBasisSet,ElemType.c_str());
+          if(atomicBasisSet<0){
+            std::cerr<<"Could not open  group EleTycBasisSet in H5; Probably Corrupt H5 file"<<std::endl;
```
What does 'EleTyc'  mean? Here it shows up in a error message a user might see.

prckent commented:
Current status: I believe this is being made MPI parallel before merging.

anbenali commented:
Indeed.. I should push the last function (SPO class with parallel) today...

On Thu, Oct 19, 2017 at 2:17 PM, Paul R. C. Kent <notifications@github.com>
wrote:

> Current status: I believe this is being made MPI parallel before merging.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/QMCPACK/qmcpack/pull/411#issuecomment-338009311>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AXl6ww1HvcIU-t20JKxgJqX-Je5Crxmaks5st6AzgaJpZM4P0wYZ>
> .
>



-- 
-----------------------
Anouar Benali, PhD
Leadership Computing Facility
Argonne National Laboratory
Building 240 Office - 2127
9700 S Cass Av., Argonne Il, 60439
(630) 252-0058

prckent commented:
@ye-luo OK to merge?

ye-luo commented:
@prckent not yet. @anbenali told me he needs to adjust some tests and I need to go through a bit.

ye-luo commented on src/QMCWaveFunctions/SPOSetBase.cpp:
```diff
@@ -174,14 +174,11 @@ bool SPOSetBase::put(xmlNodePtr cur)
   else
-      if(H5file!=true){
+      if(H5file!=true)
          APP_ABORT("Error in Opening HDF5");
```
APP_ABORT needs the surrounding bracket.

anbenali commented:
I corrected all errors, cleaned the code, added parallel read everywhere
and addressed each point.
Added are tests for the XML reader and the HDF5 reader using 12 nodes and
32 threads (done on BGQ) for less than 2 minutes (200 blocks).

there was no python script to add the test so I am wondering where to
modify the script for the molecules tests. if someone points to it I can
add it.

I do not expect to continue working on this unless there is another
request.

Cheers

Anouar

On Wed, Oct 25, 2017 at 1:32 PM, Ye Luo <notifications@github.com> wrote:

> *@ye-luo* commented on this pull request.
> ------------------------------
>
> In src/QMCWaveFunctions/SPOSetBase.cpp
> <https://github.com/QMCPACK/qmcpack/pull/411#discussion_r146947316>:
>
> > @@ -174,14 +174,11 @@ bool SPOSetBase::put(xmlNodePtr cur)
>    if(H5file==false)
>      success = putFromXML(coeff_ptr);
>    else
> -      if(H5file!=true){
> +      if(H5file!=true)
>           APP_ABORT("Error in Opening HDF5");
>
> APP_ABORT needs the surrounding bracket.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/QMCPACK/qmcpack/pull/411#pullrequestreview-71966585>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AXl6w5Hk_gLaYFc-yl2vgVSsR28KAzWWks5sv365gaJpZM4P0wYZ>
> .
>



-- 
-----------------------
Anouar Benali, PhD
Leadership Computing Facility
Argonne National Laboratory
Building 240 Office - 2127
9700 S Cass Av., Argonne Il, 60439
(630) 252-0058

prckent commented:
Thanks Anouar.

@ye-luo OK?

For testing I think we should setup 1x16,16x1, and 4x4 variants as we have for the other tests, and make the short versions 1-2 minutes in duration. No python is needed to do this. The settings go in tests/system/CMakeLists.txt . Cut & paste coding should be sufficient. Do you have the energy to do this or would you prefer to delegate?

anbenali commented:
@prckent
I will do that now. :)
Thanks for the details Paul.

On Thu, Oct 26, 2017 at 9:46 AM, Paul R. C. Kent <notifications@github.com>
wrote:

> Thanks Anouar.
>
> @ye-luo <https://github.com/ye-luo> OK?
>
> For testing I think we should setup 1x16,16x1, and 4x4 variants as we have
> for the other tests, and make the short versions 1-2 minutes in duration.
> No python is needed to do this. The settings go in
> tests/system/CMakeLists.txt . Cut & paste coding should be sufficient. Do
> you have the energy to do this or would you prefer to delegate?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/QMCPACK/qmcpack/pull/411#issuecomment-339690072>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AXl6w8KOaawg5fRNJkbtu-rYKmmXXPMrks5swJs4gaJpZM4P0wYZ>
> .
>



-- 
-----------------------
Anouar Benali, PhD
Leadership Computing Facility
Argonne National Laboratory
Building 240 Office - 2127
9700 S Cass Av., Argonne Il, 60439
(630) 252-0058

ye-luo commented:
I need to change a few things. I think now the noMPI will not build.
@anbenali do you still need to touch the code or just add tests? if it is only the test, can I work on the code?
Please also clean up the reference folder by keeping only the scalar.dat and input/output.

anbenali commented:
Ye,

I am not touching the code anymore. I will just finish the test.

On Thu, Oct 26, 2017 at 9:57 AM, Ye Luo <notifications@github.com> wrote:

> I need to change a few things. I think now the noMPI will not build.
> @anbenali <https://github.com/anbenali> do you still need to touch the
> code or just add tests? if it is only the test, can I work on the code?
> Please also clean up the reference folder by keeping only the scalar.dat
> and input/output.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/QMCPACK/qmcpack/pull/411#issuecomment-339693670>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AXl6w19Atg0JmGu-wf_wtQM_8uep4nLsks5swJ3HgaJpZM4P0wYZ>
> .
>



-- 
-----------------------
Anouar Benali, PhD
Leadership Computing Facility
Argonne National Laboratory
Building 240 Office - 2127
9700 S Cass Av., Argonne Il, 60439
(630) 252-0058

anbenali commented:
@prckent @ye-luo 

Tests updated and passed.

prckent commented:
@anbenali Excellent.

Will merge once we have @ye-luo 's fixes.

ye-luo commented:
Since this PR starts to gain weight. We need to get it merged ASAP.
Anouar has address my request of code simplification and parallel builder.
@anbenali Please address Paul's request of updating the input example of the converter.

prckent commented:
@ye-luo To update the inputs we need to generate/verify a set that works sensibly for, say, water, and also decide on a set of names. Probably best not to keep changing these in later PRs. However I take your point that this PR is "gaining weight"...

prckent commented:
However, we could just merge now and update the converter "soon".

I would prefer that (say) MSD updates are not mixed with this PR which is an argument for merging now.

@markdewing Preferences?

ye-luo commented:
We can do it by first creating a branch before the MSD change and PR it. Once that one is merged, we can update this PR and get MSD builder change in.

markdewing commented:
As long it doesn't break anything current, merging now is okay with me.   I agree with @ye-luo about the weight of this PR and merging sooner is better.

prckent commented:
Thanks Mark.

The expected "breakage" is a change in the behavior of the converter. If it proves problematic people can use the release version on an interim basis. 

Will merge once the CI has passed.

anbenali commented:
@prckent The converter is retro compatible and inputs have been modified to pass the tests (due to a change in the tag names for unification purposes). Are there any additional converter issues?
